tools/shared.py

```diff
@@ -384,7 +384,7 @@ def parse_config_file():
     JS_ENGINES = [NODE_JS]
 
   if CLOSURE_COMPILER is None:
-    CLOSURE_COMPILER = path_from_root('third_party', 'closure-compiler', 'compiler.jar')
+    CLOSURE_COMPILER = path_from_root('node_modules', '.bin', 'google-closure-compiler' + ('.cmd' if WINDOWS else ''))
```

Comment:
I feel like we should be able to use "npx google-closure-compiler" here to avoid the internal knowledge of the layout.

Reply:
`npx` has many more moving pieces, such as install-on-demand if the package does not exist. I like this explicit behavior better. Since we are not dealing with randomly updating package, and Closure is quite stable, I don't think this is much of an issue.

Reply:
OK, I agree that maybe npx is more than we want.  But once this lands i would like to find a better solution.  IIUC pocking directly into the `.bin` directory is not recommend as that is npm/node implementation detail.

I did a little research and foudn this: https://2ality.com/2016/01/locally-installed-npm-executables.html

Seems like maybe `npm bin` is the way to get access to that path.